libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1104,29 +1104,65 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	case FunctionType::Kind::ABIEncode:
 	case FunctionType::Kind::ABIEncodePacked:
 	case FunctionType::Kind::ABIEncodeWithSelector:
+	case FunctionType::Kind::ABIEncodeCall:
 	case FunctionType::Kind::ABIEncodeWithSignature:
 	{
 		bool const isPacked = functionType->kind() == FunctionType::Kind::ABIEncodePacked;
 		solAssert(functionType->padArguments() != isPacked, "");
 		bool const hasSelectorOrSignature =
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSelector ||
+			functionType->kind() == FunctionType::Kind::ABIEncodeCall ||
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSignature;
 
 		TypePointers argumentTypes;
 		TypePointers targetTypes;
 		vector<string> argumentVars;
-		for (size_t i = 0; i < arguments.size(); ++i)
+		string selector;
+		vector<ASTPointer<Expression const>> argumentsOfEncodeFunction;
+
+		if (functionType->kind() == FunctionType::Kind::ABIEncodeCall)
 		{
-			// ignore selector
-			if (hasSelectorOrSignature && i == 0)
-				continue;
-			argumentTypes.emplace_back(&type(*arguments[i]));
-			targetTypes.emplace_back(type(*arguments[i]).fullEncodingType(false, true, isPacked));
-			argumentVars += IRVariable(*arguments[i]).stackSlots();
+			solAssert(arguments.size() == 2, "");
+			auto const tupleExpression = dynamic_pointer_cast<TupleExpression const>(arguments[1]);
+			auto const tupleType = dynamic_cast<TupleExpression const*>(arguments[1].get());
+
+			solAssert(!tupleType || tupleExpression, "Tuple type without tuple expression?!");
+			// Account for tuples with one component which become that component
+			if (!tupleType || tupleExpression->isInlineArray())
+				argumentsOfEncodeFunction.push_back(arguments[1]);
+			else
+				for (auto component: tupleExpression->components())
```

Comment:
Did you test this in case the argument is not a tuple expression? This should result in a null pointer dereference.